webapp/graphite/metric_filter/models.py

```diff
@@ -0,0 +1,56 @@
+from django.db import models
+from django.contrib.auth.models import Group
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+
+class MetricFilter(models.Model):
+  id = models.AutoField(primary_key=True)
+  filter = models.CharField(max_length=256)
+  description = models.CharField(max_length=200)
+  type_filter = models.IntegerField(choices=[(0, 'include'), (1, 'exclude')])
+  __str__ = lambda self: "[%s]: %s " % (getTypeFilter(self.type_filter) ,self.filter)
+
+  class Meta:
+    db_table = "metric_filter"
+
+class UserMetricFilter(models.Model):
+  user = models.ForeignKey(User)
+  metric_filter = models.ManyToManyField(MetricFilter)
+  __str__ = lambda self: "%s" % self.use
+
+  class Meta:
+    db_table = "user_metric_filter"
+
+  @staticmethod
+  def getFilters(id):
+    try:
+      userFilter = UserMetricFilter.objects.get(user = id)
```

Comment:
Are users supposed to have at most one `UserMetricFilter`? You have a `ForeignKey` to `User` but if a user have multiple filters, calling `.get()` as done here will raise an exception.

Given that a UserMetricFilter has an m2m relation to MetricFilter, it seems that it'd be enough to make the UserMetricFilter - User relation a `OneToOneField` instead of a `ForeignKey`. Same logic for `GroupMetricFilter`.